### PR TITLE
Remove single/double precision specialization

### PR DESCRIPTION
**Pull request by andyspiros**
With these changes Serialbox no longer has the configuration variable `SINGLEPRECISION` for compile-time specialization. It just handles both single and double precision equally well.

**I suggest to tag the commit after the merge (if we are going to merge) `0.1` so to start a version numbering**.

### CODE REVIEW

**Comment by andyspiros** [comment]
The unittests work and so does the build script.
However, we should check whether the dycore uses `AsReal`, because that is the only function which really changed API. I will check that.

**Comment by andyspiros** [comment]
@cosunae @pspoerri just to notify you...

**Comment by cosunae** [comment]
launch jenkins

File: unittest/wrapper/ReadWriteUnittest.cpp [hunk 1]
```diff
@@ -13,9 +13,9 @@ class ReadWriteUnittest: public ::testing::Test {
 
     // These vectors are used as Fortran-style storage
     std::vector<int> fieldInt2, fieldInt3;
-    std::vector<Real> fieldReal1, fieldReal3;
+    std::vector<double> fieldReal1, fieldReal3;
     std::vector<int> fieldcheckInt2, fieldcheckInt3;
-    std::vector<Real> fieldcheckReal1, fieldcheckReal3;
+    std::vector<double> fieldcheckReal1, fieldcheckReal3;
```
Comment (pspoerri) [inline unittest/wrapper/ReadWriteUnittest.cpp]: I d'like to see a single precision unittest here.

**Comment by pspoerri** [comment]
How do you do the interop with fortran in this float width agnostic library?

**Comment by andyspiros** [comment]
The interop does not require communication of data type. When you register a field you give the name of the type and , most importantly, the extent of the element in bytes. When reading/writing a field you just pass around the pointer as `void*`, the strides in bytes and the name of the field, which gives the remaining information.

File: unittest/wrapper/ReadWriteUnittest.cpp (hunk 1 above)
Comment (cosunae) [inline unittest/wrapper/ReadWriteUnittest.cpp]: yes, before we were implicitly testing single precision with the matrix build, and now this is gone. 
@andyspiros could you parametrized these classes of test with a template parameter to launch double and float gtests?

**Comment by cosunae** [comment]
launch jenkins

**Comment by cosunae** [comment]
what is the status of this? I think we are still missing unittests for single precision, isnt?

**Comment by andyspiros** [comment]
We move to the MeteoSwiss-APN/serialbox#30 pull request.

**Pull request closed without merging**